src/vs/workbench/contrib/accessibility/browser/accessibleView.ts

```diff
@@ -0,0 +1,169 @@
+	) {
+		super();
+		this._editorContainer = document.createElement('div');
+		this._editorContainer.classList.add('accessible-view');
+		const codeEditorWidgetOptions: ICodeEditorWidgetOptions = {
+			contributions: EditorExtensionsRegistry.getSomeEditorContributions([LinkDetector.ID, SelectionClipboardContributionID, 'editor.contrib.selectionAnchorController'])
+		};
+		const editorOptions: IEditorConstructionOptions = {
+			...getSimpleEditorOptions(),
+			lineDecorationsWidth: 6,
+			dragAndDrop: true,
+			cursorWidth: 1,
+			wrappingStrategy: 'advanced',
+			wrappingIndent: 'none',
+			padding: { top: 2, bottom: 2 },
+			quickSuggestions: false,
+			renderWhitespace: 'none',
+			dropIntoEditor: { enabled: true },
+			accessibilitySupport: this._configurationService.getValue<'auto' | 'off' | 'on'>('editor.accessibilitySupport'),
+			cursorBlinking: this._configurationService.getValue('terminal.integrated.cursorBlinking'),
+			readOnly: true
+		};
+		this._editorWidget = this._register(this._instantiationService.createInstance(CodeEditorWidget, this._editorContainer, editorOptions, codeEditorWidgetOptions));
+	}
+
+	show(provider: IAccessibleContentProvider): void {
+		const delegate: IContextViewDelegate = {
+			getAnchor: () => this._editorContainer,
+			render: (container) => {
+				return this._render(provider, container);
+			},
+			onHide: () => {
+				provider.onClose();
+			}
+		};
+		this._contextViewService.showContextView(delegate);
+	}
+
+	private _render(provider: IAccessibleContentProvider, container: HTMLElement): IDisposable {
+
+		const fragment = 'Exit this menu via the Escape key.\n' + provider.provideContent();
+
+		this._getTextModel(URI.from({ path: `accessible-view-${provider.id}`, scheme: 'accessible-view', fragment })).then((model) => {
+			if (!model) {
+				return;
+			}
+			this._editorWidget.setModel(model);
+			const domNode = this._editorWidget.getDomNode();
+			if (!domNode) {
+				return;
+			}
+			container.appendChild(domNode);
+			this._layout();
+			this._register(this._editorWidget.onKeyDown((e) => {
+				if (e.keyCode === KeyCode.Escape) {
+					this._contextViewService.hideContextView();
+				}
+			}));
+			this._register(this._editorWidget.onDidBlurEditorText(() => this._contextViewService.hideContextView()));
+			this._editorWidget.updateOptions({ ariaLabel: localize('accessible-view-label', "{0}", provider.options.ariaLabel) });
+			this._editorWidget.focus();
+		});
+		return { dispose: () => { provider.onClose(); } } as IDisposable;
+	}
+
+	private _layout(): void {
+		const domNode = this._editorWidget.getDomNode();
+		if (!domNode) {
+			return;
+		}
+
+		const windowWidth = window.innerWidth;
+		const windowHeight = window.innerHeight;
+
+		const width = windowWidth * .4;
+		let height = windowHeight * .4;
+		this._editorWidget.layout({ width, height });
+		height = this._editorWidget.getContentHeight();
+		this._editorWidget.layout({ width, height });
```

Comment:
Is this double layout needed? You might get away with just fetching the actual height and not doing another layout?

Reply:
the actual height is 0 or something really small initially. that's why this is needed

Reply:
That doesn't sound right, you're laying out based on `window.innerHeight` so it should be getting a good value on the first call. @rebornix any idea what's wrong?

Reply:
Yeah I wish there was a way for the editor to know the correct `layout` dimensions based on the content

Reply:
Since I couldn't figure out how to make that work, I set the dimensions and then get the height so it precisely fits the content but has the same width

Reply:
I'm not sure why a double layout is needed. We only need the second one if the initial height `height = windowHeight * .4;` is only an estimation but not a target height. Maybe we sync offline to go through this together.

Reply:
we need to call layout on content height change

Reply:
we should make sure layout isn't called too many times, but going to defer that